534: accessibility navigation issue fix for manage profile and choose…

meghac538 opened the pull request:
PR created for #534 
Updated flatListData to maintain original order of profile records
Refactored ManageProfilesScreen and ProfileSelectionScreen to remove inverted FlatList and improve layout structure
Added listContainer style for better flexibility in rendering 

**Before Update:**
<img width="1183" height="1050" alt="Screenshot 2025-09-13 at 11 46 45 AM" src="https://github.com/user-attachments/assets/0546ba83-7b93-4113-bf86-b5688efb2305" />
<img width="1183" height="1050" alt="Screenshot 2025-09-13 at 11 48 03 AM" src="https://github.com/user-attachments/assets/148d0047-4881-46c3-a9f7-a5ab90f54d36" />

**After Update:**
<img width="1183" height="1050" alt="Screenshot 2025-09-13 at 11 41 55 AM" src="https://github.com/user-attachments/assets/d6d38efb-1c64-43f1-af01-aaa8a9d5b265" />
<img width="1183" height="1050" alt="Screenshot 2025-09-13 at 11 43 48 AM" src="https://github.com/user-attachments/assets/8b51e514-4ec6-4ff7-83db-fa9d7e2accc0" />

alexfigtree reviewed (changes requested):
Now that you have lcov-reports generating, could you please start adding/filling out tests for the components you work on to help us improve code coverage?

alexfigtree commented:
@meghac538 Also, Dmitri made a comment on the original issue but we currently don't have documentation on how to turn on accessibility features for iOS/Android. Were you able to test on a screen reader to ensure that the order of items is still read in the same way we are displaying?

alexfigtree commented:
> Thanks @alexfigtree! I’ve added tests for the updated components and also attached screen recordings for accessibility testing.
> 
> **iOS:** Verified with VoiceOver — accessibility order looks good both before and after the changes. https://github.com/user-attachments/assets/29ecfbf8-b98b-47a9-968a-46d8484dbe24
> 
> **Android**: Added recordings showing TalkBack behavior before and after applying the fix in this PR. **Before:** https://github.com/user-attachments/assets/985fb1e3-21ac-45e5-824a-5901b00468ad
> 
> **After** https://github.com/user-attachments/assets/64475743-e22c-4526-bba5-684205ece3d8
> 
> Would you like me to also add documentation on how to enable accessibility features on iOS/Android?

@meghac538 Some more documentation would be great! I'm in the process of revising our README. If you can post a blurb here on how to enable accessibility, I can add it to the README. Or you're welcome to edit the README as it currently and add that change to this PR.

meghac538 commented:
> @meghac538 Some more documentation would be great! I'm in the process of revising our README. If you can post a blurb here on how to enable accessibility, I can add it to the README. Or you're welcome to edit the README as it currently and add that change to this PR.

@alexfigtree I’ve updated the README with the following:

- Included accessibility features for both iOS and Android
- Added detailed instructions for enabling accessibility features
- Outlined accessibility testing methods for developers
- Added development guidelines to ensure accessibility compliance

alexfigtree commented:
@meghac538 Readme looks great, thank you for that. As for the new tests, standard lcov coverage should reach 85%. You can look at your statements to see how you can bump this up with more tests.

meghac538 commented:
> @meghac538 Readme looks great, thank you for that. As for the new tests, standard lcov coverage should reach 85%. You can look at your statements to see how you can bump this up with more tests.

Thanks @alexfigtree! I’ve updated the tests to bring the coverage up to 85%

Changes include:
- Introduced a new file mock for testing purposes.
- Enhanced the Cache class to handle loading errors gracefully.
- Implemented biometric keychain functions with comprehensive tests for storing, retrieving, and resetting credentials.
- Added tests for credential validity period and dynamic styles.
- Created utility tests for error handling, text formatting, and time functions.
- Updated ProfileItem and ManageProfilesScreen tests for improved interaction and coverage.

Next, we can continue improving coverage for other components and functionalities as we work on them

meghac538 commented:
> @meghac538 I see Santhosh has completed work for this, so this PR can now be merged.

Thanks for the heads up, Alex! The merge has been completed.
